app/src/main/java/ingsw/group1/findmyphone/log/LogManager.java

```diff
@@ -0,0 +1,157 @@
+     * {@link LogManager#setSortingOrder(EventOrder)} and {@link LogManager#filter(String)}.
+     */
+    private FilterableList<String, LogItem> itemsView;
+
+    private RecyclerView.Adapter currentListener;
+    private SMSLogDatabase targetDatabase;
+    private LogItemFormatter itemFormatter;
+    private String currentQuery = "";
+    private EventOrder currentOrder = EventOrder.NEWEST_TO_OLDEST;
+
+    /**
+     * Default constructor.
+     *
+     * @param targetDatabase The database this Manager should listen to.
+     */
+    public LogManager(@NonNull SMSLogDatabase targetDatabase,
+                      @NonNull LogItemFormatter itemFormatter) {
+        this.targetDatabase = targetDatabase;
+        this.itemFormatter = itemFormatter;
+        init();
+    }
+
+    /**
+     * Method initializing this instance, retrieving the data from the database and formatting it.
+     */
+    private void init() {
+        allItems = new FilterableList<>(itemFormatter.formatItems(targetDatabase.getAllEvents()));
+        Collections.sort(allItems, LogItemComparatorFactory.newComparator(currentOrder));
+        itemsView = new FilterableList<>(allItems);
+    }
+
+    /**
+     * Method to return the item that should be displayed at the {@code position} position in a
+     * {@link android.widget.ListView}/{@link RecyclerView}.
+     *
+     * @param position The position of the item.
+     * @return The item that should be shown at the given position.
+     */
+    public LogItem getItem(int position) {
+        return itemsView.get(position);
+    }
+
+    /**
+     * Method to get the displayable item count.
+     *
+     * @return The number of items in {@link LogManager#itemsView}.
+     */
+    public int count() {
+        return itemsView.size();
+    }
+
+    /**
+     * Method to get a shallow copy of the items view.
+     *
+     * @return A shallow copy of {@link LogManager#itemsView}.
+     */
+    public List<LogItem> getAll() {
+        return new ArrayList<>(itemsView);
+    }
+
+    /**
+     * Method to set a new listener for this Manager.
+     *
+     * @param newListener The new Adapter that should get notified of changes.
+     */
+    public void setListener(@NonNull RecyclerView.Adapter newListener) {
+        currentListener = newListener;
+    }
+
+    /**
+     * Method resetting the current listener to {@code null}.
+     */
+    public void removeListener() {
+        currentListener = null;
+    }
+
+    /**
+     * Method to notify the current listener.
+     */
+    private void notifyListener() {
+        if (currentListener != null)
+            currentListener.notifyDataSetChanged();
+    }
+
+    /**
+     * This method sorts the items and notifies the listener.
+     *
+     * @param newSortingOrder The sorting order to use.
+     */
+    public void setSortingOrder(EventOrder newSortingOrder) {
+        if (newSortingOrder == currentOrder) return;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5010214)